sbi/inference/posteriors/direct_posterior.py

```diff
@@ -240,6 +243,7 @@ def sample(
         sample_with_mcmc: Optional[bool] = None,
         mcmc_method: Optional[str] = None,
         mcmc_parameters: Optional[Dict[str, Any]] = None,
+        max_sampling_batch_size: int = 1000,
```

Comment:
For consistency, I would make it a `dict` and call it `rejection_sampling_parameters`. Then, we also make them a property and give them a setter, as [here](https://github.com/mackelab/sbi/blob/da0daf5fee8da0a055b37ac639690342441a9d32/sbi/inference/posteriors/base_posterior.py#L145)

If we do not want this, then it should be reflected in the name that this refers only to rejection_sampling. And explicitly write in the docstring that it is ignored for `sample_with_mcmc=True`.

also @jan-matthis what do you think?

Reply:
`rejection_sampling_parameters` as a dict sounds good to me 👍

Reply:
Should we move up the `rejection_sampling_parameters` all the way to `base_inference` like the `mcmc_parameters`?

Instead we could offer to only pass via `posterior.sample()`

Reply:
Hm, I would define them in the `__init__` of the direct posterior class because the posteriors won't allow rejection sampling